src/cargo/core/compiler/fingerprint/mod.rs

```diff
@@ -1860,11 +1860,18 @@ where
     for path in paths {
         let path = path.as_ref();
 
-        // Assuming anything in cargo_home is immutable (see also #9455 about marking it readonly)
-        // which avoids rebuilds when CI caches $CARGO_HOME/registry/{index, cache} and
-        // $CARGO_HOME/git/db across runs, keeping the content the same but changing the mtime.
-        if let Ok(true) = home::cargo_home().map(|home| path.starts_with(home)) {
-            continue;
+        // Assuming anything in cargo_home/{git, registry} is immutable
+        // (see also #9455 about marking cargo_home readonly) which avoids rebuilds when CI caches
```

Comment:
FWIW: #9455 is not about marking CARGO_HOME readonly, but rather only a subdirectory of it.